src/async_client.cpp

```diff
@@ -304,7 +305,8 @@ itoken_ptr async_client::disconnect(long timeout, void* userContext, iaction_lis
 	tok->set_action_callback(cb);
 	add_token(tok);
 
-	disconnect_options opts(int(timeout), dynamic_cast<token*>(tok.get()));
+	// TODO may truncate timeout
+	disconnect_options opts(static_cast<int>(timeout), dynamic_cast<token*>(tok.get()));
```

Comment:
I'm considering replacing all of the integer timeout values with std::chrono::duration types. But in the meantime, yes, you're correct on this point.